src/main/java/de/tum/in/www1/exerciseapp/service/QuizExerciseService.java

```diff
@@ -110,6 +114,36 @@ public QuizExercise findOne(Long id) {
         return quizExerciseRepository.findOne(id);
     }
 
+    /**
+     * Get one quiz exercise by id.
+     *
+     * @param id the id of the entity
+     * @return the entity
+     */
+    @Transactional(readOnly = true)
+    public QuizExercise findOneWithQuestionsAndStatistics(Long id) {
+        log.debug("Request to get Quiz Exercise : {}", id);
+        return quizExerciseRepository.findOneByIdWithEagerQuestionsAndStatistics(id);
+    }
+
+    /**
+     * Get one quiz exercise by id.
+     * NOTE: This method is to be used by Threads outside Spring (e.g. the Timer when a Quiz ends)
```

Comment:
How is it different and why is there a specific method for it? Please add an explanation